core/src/main/scala/io/snappydata/SnappyStoreTableSizeProviders.scala

```diff
@@ -0,0 +1,188 @@
+
+        result.asInstanceOf[ArrayList[HashMap[String, Long]]]
+            .asScala.foreach(_.asScala.foreach(row => {
+          var totalSize: Long = row._2
+          if (currentTableSizeInfo.contains(row._1)) {
+            totalSize = currentTableSizeInfo.get(row._1).get + row._2
+          }
+          currentTableSizeInfo.put(row._1, totalSize)
+        }))
+
+          tableSizeInfo = currentTableSizeInfo.toMap
+      }
+
+      override def getLoggerI18n: LogWriterI18n = {
+        return Misc.getGemFireCache.getLoggerI18n
+      }
+    }
+  }
+
+  def getTableSize(tableName: String, isColumnTable: Boolean = false):
+  Option[Long] = {
+    val currentTableSizeInfo = tableSizeInfo
+    if (currentTableSizeInfo == null || !currentTableSizeInfo.contains(tableName)) {
+      None
+    }
+    else {
+      if (isColumnTable) {
+        val optStat = currentTableSizeInfo.get(ColumnFormatRelation.cachedBatchTableName(tableName))
+        if (optStat.isDefined) {
+          Some(optStat.get + currentTableSizeInfo.get(tableName).get)
+        } else {
+          None
+        }
+      } else {
+        Some(currentTableSizeInfo.get(tableName).get)
+      }
+    }
+  }
+
+}
+
+object StoreTableSizeProvider {
+
+  def getTableSizes: Seq[UIAnalytics] = {
+    val currentTableStats = tryExecute(conn => getMemoryAnalyticsDetails(conn))
+    if (currentTableStats == null) {
+      return Seq.empty
+    }
+    val internalColumnTables = currentTableStats.filter(entry => isColumnTable(entry._1))
+
+    var columnTableNames = scala.collection.mutable.Seq[String]()
+
+    (internalColumnTables.map(entry => {
+      val rowBuffer = getRowBufferName(entry._1)
+      val rowBufferSize = currentTableStats.get(rowBuffer).get
+      columnTableNames = columnTableNames.+:(rowBuffer).+:(entry._1)
+      new UIAnalytics(rowBuffer, rowBufferSize, entry._2, true)
+    })
+        ++ currentTableStats.filter(
+      entry => {
+        !columnTableNames.contains(entry._1)
+      }).map(rowEntry =>
+      new UIAnalytics(rowEntry._1, rowEntry._2, 0, false))).toSeq
+  }
+
+
+  private def getMemoryAnalyticsDetails(conn: Connection): mutable.Map[String, Long] = {
+    val currentTableStats = mutable.Map[String, Long]()
+    val stmt = "select TABLE_NAME," +
+        "SUM(TOTAL_SIZE)" +
+        "from SYS.MEMORYANALYTICS " +
+        "WHERE table_name not like 'HIVE_METASTORE%'  group by TABLE_NAME"
+    val rs = conn.prepareStatement(stmt).executeQuery()
+    while (rs.next()) {
+      val name = rs.getString(1)
+      val totalSize = convertToBytes(rs.getString(2))
+      currentTableStats.put(name, totalSize)
+    }
+    currentTableStats
+  }
+
+  private def convertToBytes(value: String): Long = {
+    if (value == null) 0 else (value.toDouble * 1024).toLong
+  }
+
+
+  private def getRowBufferName(columnStoreName: String): String = {
+    columnStoreName.replace(SHADOW_TABLE_SUFFIX, "").
+        replace(INTERNAL_SCHEMA_NAME, "").
+        replaceFirst(".", "").
```

Comment:
Perhaps I am missing something here. The format of column table name is as you say, then where is a "." there which this replaceFirst is trying to remove?